sycl/test-e2e/ESIMD/unified_memory_api/Inputs/gather.hpp

```diff
@@ -286,3 +286,238 @@ template <typename T, TestFeatures Features> bool testUSM(queue Q) {
   }
   return Passed;
 }
+
+template <typename T, uint16_t N, uint16_t VS, bool UseMask, bool UsePassThru,
+          bool UseProperties, typename PropertiesT>
+bool testSLM(queue Q, uint32_t MaskStride, PropertiesT) {
```

Comment:
Is it possible to add compile-time tests to memory_properties.cpp for all the possible combinations? We have been doing that to make sure simd_view works with everything.